README.md

```diff
@@ -1,313 +1,358 @@
-```
-### Handling the logs
+* `nsq` : The central location where logs are sent by `collector`(s) after formatting as messages.
 
-- To get the nginx logs in json format we have to modify the `nginx.conf` file in `/etc/nginx` directory.
+* `forwarder` : The program that runs on the central node which receives messages from `nsq` and passes it on to `forwarders`
+* `forwarders` : The parsers that take messages and formats it for storing at `target`(s) databases
+* `targets` : The databases that store the logs finally so that we can query on them easily
 
-- Modify the `Logging Settings` section in the `nginx.conf` file by adding the below lines:
-```bash
-log_format  json  '{'
-                    '"remote_addr": "$remote_addr",'
-                    '"remote_user": "$remote_user",'
-                    '"timestamp": "$msec",'
-                    '"request": "$request",'
-                    '"status": "$status",'
-                    '"request_time": "$request_time",'
-                    '"body_bytes_sent": "$body_bytes_sent",'
-                    '"http_referer": "$http_referer",'
-                    '"http_user_agent": "$http_user_agent",'
-                    '"http_x_forwarded_for": "$http_x_forwarded_for",'
-                    '"upstream_response_time": "$upstream_response_time"'
-                    '}';
-```
 
-- Modify the line: `access_log /var/log/nginx/access.log;` to `access_log /var/log/nginx/access.log json;`
-
-- To test nginx config file
-```bash 
-nginx -t
-```
+----------
 
-- It shows like:
-```bash
-nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
 
-nginx: configuration file /etc/nginx/nginx.conf test is successful
-```
+## Features
 
-- Restart the nginx by using command:
-```bash
-/etc/init.d/nginx restart
-```
+* Guaranteed delivery of each log line from files to `targets`
+* Reduced latency between a log being generated an being present in the `targets`
+* Options to add custom `formatters` & `target` databases
+* File poll if log file not yet generated
+* Works on rotational log files
+* Output format of processed log lines (dictionary)
+    * `id` (str) - A unique id per log with time ordering. Useful to avoid storing duplicates.
+    * `timestamp` (str) - ISO Format time. eg:
+    * `data` (dict) - Parsed log data
+    * `raw` (str) - Raw log line read from the log file
+    * `host` (str) - Hostname of the node where this log was generated
+    * `formatter` (str) - name of the formatter that processed the raw log line
+    * `file` (str) - Full path of the file in the host where this log line came from
+    * `type` (str) - One of "log", "metric" (Is there one more type?)
+    * `level` (str) - Log level of the log line.
+    * `event` (str) - LOG event
+    * `error` (bool) - True if collection handler failed during processing
+    * `error_tb` (str) - Error traceback
 
-### Usage
 
-- After installation of the logagg module, we can perform operations in the python shell.
+---
```

Comment:
@supriyopaul  These hyphens looks odd. Don't believe they are required. Also, for the above fields, example values might help the user understand them better.